src/System.Runtime.Caching/tests/System.Runtime.Caching/ObjectCacheTest.cs

```diff
@@ -0,0 +1,148 @@
+//
+// Permission is hereby granted, free of charge, to any person obtaining
+// a copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+// 
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.Caching;
+
+using Xunit;
+using MonoTests.Common;
+
+namespace MonoTests.System.Runtime.Caching
+{
+    public class ObjectCacheTest
+    {
+        [Fact]
+        [ActiveIssue(25168)]
+        private static void Host_SetToProvider()
+        {
+            var tns1 = new TestNotificationSystem();
+            var tns2 = new TestNotificationSystem();
+            ObjectCache.Host = tns1;
+            Assert.NotNull(ObjectCache.Host);
+            Assert.Equal(tns1, ObjectCache.Host);
+
+            Assert.Throws<InvalidOperationException>(() =>
+            {
+                ObjectCache.Host = tns2;
+            });
+        }
+
+        [Fact]
+        public void Add_CacheItem_CacheItemPolicy()
+        {
+            var poker = new PokerObjectCache();
+            bool ret;
+
+            ret = poker.Add(null, null);
+            Assert.True(ret);
+            Assert.Equal("AddOrGetExisting (CacheItem value, CacheItemPolicy policy)", poker.MethodCalled);
+
+            var item = new CacheItem("key", 1234);
+            ret = poker.Add(item, null);
+            Assert.True(ret);
+            Assert.Equal("AddOrGetExisting (CacheItem value, CacheItemPolicy policy)", poker.MethodCalled);
+
+            ret = poker.Add(item, null);
+            Assert.False(ret);
+            Assert.Equal("AddOrGetExisting (CacheItem value, CacheItemPolicy policy)", poker.MethodCalled);
+        }
+
+        [Fact]
+        public void Add_String_Object_CacheItemPolicy_String()
+        {
+            var poker = new PokerObjectCache();
+            bool ret;
+
+            ret = poker.Add(null, null, null, null);
+            Assert.True(ret);
+            Assert.Equal("AddOrGetExisting (string key, object value, CacheItemPolicy policy, string regionName = null)", poker.MethodCalled);
+
```

Comment:
I don't like duplicating constants in multiple places in code. Can you refactor? Maybe pull the string out to be a constant declared immediately before the method declaration which can be referenced here.

Reply:
will address this (and the one above) in #26171